src/proof.rs

```diff
@@ -25,15 +25,15 @@ use primitives::H256;
 #[derive(Clone, Eq, PartialEq, Debug)]
 pub struct CryptoProofUnit {
     pub root: H256,
-    pub key: H256,
+    pub key: Bytes,
```

Comment:
Why did you change H256 to Bytes?

Reply:
Because all other interface of trie takes before-hash keys

Reply:
Then you need to specify the reason in the commit title or commit message.
I can't find out what did you do in the commit message: "Organize interface of proof"

Also, please split this commit into two commits.
One is changing the key type from H256 to Bytes.
The other one is implementing `CryptoStructure` for the TrieDBMut and make Trie extend CryptoStructure.

They are totally different thing and included in this commit.

Reply:
ok

Reply:
Can I make 2 commits in **a single PR**? (each commit will have detailed explanation) @majecty

Reply:
Sure.

The main commit will be "implementing the `CryptoStructure` for the TrieDBMut".
The commit that changes key type is preparation work for it.